src/LiveComponent/assets/dist/live_controller.js

```diff
@@ -2393,7 +2393,8 @@ class LoadingPlugin {
     }
     getLoadingDirectives(element) {
         const loadingDirectives = [];
-        let matchingElements = element.querySelectorAll('[data-loading]');
+        let matchingElements = [...element.querySelectorAll('[data-loading]')];
+        matchingElements = matchingElements.filter((elt) => elt.closest('[data-controller~="live"]') === element);
```

Comment:
I am not sure about the use of the `closest` function. `closest` finds only one element, but in a live component you can have multiple elements with a loading attribute in nested element

Reply:
closest is an ascending selector, it look the first parent "live controller"